src/derivative_utils.jl

```diff
@@ -375,15 +361,16 @@ function calc_W!(W, integrator, nlsolver::Union{Nothing,AbstractNLSolver}, cache
   alg = unwrap_alg(integrator, true)
   mass_matrix = integrator.f.mass_matrix
   is_compos = integrator.alg isa CompositeAlgorithm
-  isnewton = alg isa NewtonAlgorithm
 
   # handle Wfact
   if W_transform && DiffEqBase.has_Wfact_t(f)
     f.Wfact_t(W, u, p, dtgamma, t)
+    isnewton(nlsolver) && set_W_γdt!(nlsolver, dtgamma)
```

Comment:
interesting... we didn't set it before?

Reply:
We used to use `gamma`, which is a constant for SDIRKs, but it was not suitable for BDF.